pkgs/development/python-modules/haystack-ai/default.nix

```diff
@@ -1,11 +1,19 @@
 {
   lib,
   buildPythonPackage,
-  fetchFromGitHub,
+  fetchPypi,
   pythonRelaxDepsHook,
+
+  # Feature flags
+  withJax ? false,
+  withTorch ? true,
+  withTensorflow ? false,
```

Comment:
The python module does not allow such flags.
This is because it causes unnecessary rebuilds and is uncontrollable by downstream packages.
This package does not need to provide anything, and users should add any backend they want to their environment.

Reply:
What do you mean, there are other python modules using `withFlag` or `enableFlag` in the tree. Do you believe it is preferable to do `overrideAttrs` to manipulate `propagatedBuildInputs` in your python environment, instead of regular `override`. To me that seems like more work, or is this now some policy in nixpkgs to which I'm unaware?

Reply:
> there are other python modules using withFlag or enableFlag in the tree.

They are used as applications or are old or have not received proper review.

> Do you believe it is preferable to do overrideAttrs to manipulate propagatedBuildInputs in your python environment, instead of regular override.

We have a function called `overridePythonAttrs` that is used to properly handle dependencies.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#overriding-python-packages-overriding-python-packages

For this package, you don't even need overridePythonAttrs, just add the necessary backends to your python environment.